### PR TITLE
Evidence provision fees

### PR DESCRIPTION
**Pull request by colinbruce**
[Pivotal ticket](https://www.pivotaltracker.com/story/show/155067395)
The EPF has only two allowable amounts that are claimable. 45 and 90 (ex VAT). CCLF enforces this logic.

**As a litigator**
**When** i am claiming an evidence provision fee
**I should** be presented only with the valid amounts that are claimable
**So** i do not make claims for erroneous amounts

### CODE REVIEW

**Comment by colinbruce** [comment]
[finishes ##155067395]

**Comment by colinbruce** [comment]
[finish #155067395]

File: spec/validators/fee/misc_fee_validator_spec.rb [hunk 1]
```diff
@@ -45,6 +45,36 @@
       end
     end
 
+    context 'when misc fee is an evidence provision fee' do
+      let(:fee_type) { build :misc_fee_type, :mievi }
+
+      before { allow(fee).to receive(:fee_type).and_return(fee_type) }
+
+      context 'when the amount is set to' do
```
Comment (jsugarman) [inline spec/validators/fee/misc_fee_validator_spec.rb]: if you put a `subject { fee }` here, you could use `is_expected.to be_valid` below, but either way

File: spec/validators/fee/misc_fee_validator_spec.rb [hunk 2]
```diff
@@ -45,6 +45,36 @@
...
+            let(:amount) { value }
+
+            it { expect(fee).to be_valid }
+          end
+        end
+
+        context 'a decimal' do
+          let(:amount) { '45.10' }
+
+          it { should_error_if_equal_to_value(fee, :amount, amount, 'incorrect_epf') }
```
Comment (jsugarman) [inline spec/validators/fee/misc_fee_validator_spec.rb]: just a suggestion, for consistency more than anything, you could get rid of the `context` and `let` and just do

```
it 'should error for decimals' do
  should_error_if_equal_to_value(fee, :amount, '45.10', 'incorrect_epf')
end
```

File: spec/validators/fee/misc_fee_validator_spec.rb [hunk 3]
```diff
@@ -45,6 +45,36 @@
...
+        end
+
+        context 'zero' do
+          let(:amount) { '0' }
+
+          it { should_error_if_equal_to_value(fee, :amount, amount, 'incorrect_epf') }
```
Comment (jsugarman) [inline spec/validators/fee/misc_fee_validator_spec.rb]: as  comment above

File: spec/factories/fee_types.rb [hunk 4]
```diff
@@ -129,6 +129,14 @@
         unique_code 'MIUPL'
         quantity_is_decimal true
       end
+
+      trait :mievi do
+        lgfs
+        description 'Evidence provision fee'
+        code 'XEVI'
+        unique_code 'MIEVI'
+        quantity_is_decimal true
```
Comment (jsugarman) [inline spec/factories/fee_types.rb]: `quantity_is_decimal ` should be false for this fee type

**Review by jsugarman — APPROVED** [review]
All good aside from some spec'ing suggestions. Also, as discussed, I think some api release documentation should go in at the same time [as here on sandbox](https://api-sandbox-adp.dsd.io/api/release_notes)

File: spec/factories/fee_types.rb (hunk 4 above)
Comment (colinbruce) [reply spec/factories/fee_types.rb]: Good catch! copy-paste error :(

**Pull request merged**